src/components/DBSearchResults.vue

```diff
@@ -433,9 +443,45 @@ export default {
     },
     genreSearch (genre) {
       return this.allMediaAsArray.filter((entry) => {
-        const genres = this.topStructure(entry).genres.map((genre) => genre.name.toLowerCase());
+        const movieEntry = this.topStructure(entry);
+        if (movieEntry.genres) {
+          const genres = this.topStructure(entry).genres.map((genre) => genre.name.toLowerCase());
+          return genres.includes(genre);
+        }
+        return false;
+      })
+    },
+    directorSearch (directors) {
+      return this.allMediaAsArray.filter((entry) => {
+        const crew = this.topStructure(entry).crew;
+        const matches = crew.filter((crew) => crew.job === 'Director');
+        const names = matches.map((match) => match.name.toLowerCase());
+        return directors.every((director) => names.includes(director));
+      })
+    },
+    tagSearch (tags) {
+      return this.allMediaAsArray.filter((entry) => {
+        const ratings = entry.ratings;
+        const ratingTags = ratings.reduce((acc, rating) => {
+          if (rating.tags) {
+            Object.values(rating.tags).forEach(tag => {
+              if (tag.title) {
+                acc.push(tag.title.toLowerCase());
+              }
+            });
+          }
+          return acc;
+        }, []);
+
+        let allTags = ratingTags;
+        
+        if (this.topStructure(entry).tags) {
+          const movieTags = this.topStructure(entry).tags;
+          const tagNames = movieTags.map((movieTag) => movieTag.title.toLowerCase());
+          allTags = ratingTags.concat(tagNames);
+        }
```

Comment:
```suggestion
        let allTags = entry.ratings.reduce((acc, rating) => {
          if (rating.tags) {
            Object.values(rating.tags).forEach(tag => {
              if (tag.title) {
                acc.push(tag.title.toLowerCase());
              }
            });
          }
          return acc;
        }, []);
        
        if (this.topStructure(entry).tags) {
          const movieTags = this.topStructure(entry).tags;
          const tagNames = movieTags.map((movieTag) => movieTag.title.toLowerCase());
          allTags = ratingTags.concat(tagNames);
        }
```
Two things here. One is this small change just to take out the extra variables. The second one I'll add in my next comment.

Reply:
I think we're going to cause trouble here with tvShows. The structure of the ratings key on a tvShow is different. Because instead of having a bunch of individual ratings there's one rating for the show and then a separate array of episode ratings. This means that if you try to run this code with a TV Show it will error and say "no method `reduce` on `ratings`"

I think the easiest way around this is to have a guard clause (a conditional at the top of this method) that uses the `currentLogIsTVLog` computed value to make this work differently for TV. Something like:
```if (currentLogIsTVLog) {
}

Reply:
I think that I am not allowed to commit your suggestion. When I try, it says it cannot do it and down below I see "1 review requesting changes by reviewers with write access." I may not have write access? Or I may not understand what I need to do. I made the changes on my machine (and fixed something in your suggested change), but if I want to commit them I need to make a new branch and do a new pull request with that branch, which doesn't seem like what I am supposed to be doing. Even so, I am going to do that anyway so nothing gets lost, and hopefully we can untangle the web later.